ldelf/ftrace.c

```diff
@@ -72,9 +92,22 @@ void ftrace_copy_buf(void *pctx, void (*copy_func)(void *pctx, void *b,
 		struct ta_elf *elf = TAILQ_FIRST(&main_elf_queue);
 		size_t dump_size = fbuf->buf_off - fbuf->head_off +
 				   fbuf->curr_size;
+		char *hstart = (char *)fbuf + fbuf->head_off;
+		char *end = hstart + dump_size;
+		char *cstart = (char *)fbuf + fbuf->buf_off;
+		char *ccurr = cstart +
+				fbuf->curr_idx * sizeof(struct __ftrace_elem);
 
 		assert(elf && elf->is_main);
-		copy_func(pctx, (char *)fbuf + fbuf->head_off, dump_size);
+
+		/* Header */
+		copy_func(pctx, hstart, fbuf->buf_off - fbuf->head_off);
+		if (fbuf->curr_size == fbuf->max_size) {
```

Comment:
Here the check for overflow flag would be self describing as compared to current one.